src/headers/SiteHeader.tsx

```diff
@@ -51,33 +51,43 @@ export interface SiteHeaderProps {
 }
 
 const SiteHeader = (props: SiteHeaderProps) => {
-  const [activeMenus, setActiveMenus] = useState<string[]>([])
+  const [activeMenu, setActiveMenu] = useState<string | null>()
   const [activeMobileMenus, setActiveMobileMenus] = useState<string[]>([])
   const [isDesktop, setIsDesktop] = useState(true)
   const [mobileDrawer, setMobileDrawer] = useState(false)
   const [mobileMenu, setMobileMenu] = useState(false)
 
+  const [navbarClass, setNavbarClass] = useState("site-header__navbar-inline")
+
+  const updateNavbarClass = () => {
+    // If the links have flex-wrapped onto the next line, apply the background color
+    const logoOffset = document.getElementById("site-header-logo")?.offsetLeft
+    const linksOffset = document.getElementById("site-header-links")?.offsetLeft
+    if (linksOffset === undefined || logoOffset === undefined) return
+    return linksOffset === 0 || linksOffset === logoOffset
+      ? setNavbarClass("site-header__navbar-wrapped")
+      : setNavbarClass("site-header__navbar-inline")
+  }
+
   const { LinkComponent } = useContext(NavigationContext)
 
   const DESKTOP_MIN_WIDTH = props.desktopMinWidth || 767 // @screen md
   // Enables toggling off navbar links when entering mobile
-  useEffect(() => {
-    if (window.innerWidth > DESKTOP_MIN_WIDTH) {
-      setIsDesktop(true)
-    } else {
-      setIsDesktop(false)
-    }
-
+  useLayoutEffect(() => {
```

Comment:
Is `useLayoutEffect` rather than `useEffect` needed because of the updateNavbarClass adding class names?

Reply:
It's needed because we're getting the offset of an element and we need the browser to finish its first paint to get an accurate number